dex/src/main/scala/com/wavesplatform/dex/model/OrderBook.scala

```diff
@@ -59,14 +65,15 @@ case class OrderBook private (bids: Side, asks: Side, lastTrade: Option[LastTrad
     submitted: AcceptedOrder,
     eventTs: Long,
     getMakerTakerFee: (AcceptedOrder, LimitOrder) => (Long, Long),
+    getOrderExecutedTs: (Long, Long) => Long,
     tickSize: Long = MatchingRule.DefaultRule.tickSize
   ): OrderBookUpdates = {
     val events = Queue(OrderAdded(submitted, OrderAddedReason.RequestExecuted, eventTs))
-    if (submitted.order.isValid(eventTs)) doMatch(eventTs, tickSize, getMakerTakerFee, submitted, events, this)
+    if (submitted.order.isValid(eventTs)) doMatch(eventTs, tickSize, getMakerTakerFee, submitted, events, this, getOrderExecutedTs)
```

Comment:
Not sure, but maybe it should be checked with `orderExecutedTs` instead of  `eventTs`, to prevent invalid exchange transactions?